### DISCUSSION TITLE
Configuring Syntax Highlighting in Code Blocks within Markdown

### DISCUSSION THREAD

**Discussion by wpalace** [LazyVim/LazyVim#4537] [Q&A]
How does syntax highlighting for code blocks within markdown files work and how do I configure it for other languages? Some languages like bash, javascript, and typescript seem to work out of the box. Other languages like sql, python, and terraform work when installed as extras. I am trying to add support for PowerShell and Cyper, which are not available as extras, so I installed them with Mason, but the syntax highlighting is not working. 

If I open a .ps1 file, the syntax highlighting for PowerShell works; but it does not work within a markdown code block. 

![image](https://github.com/user-attachments/assets/73f07a7e-350d-40a6-bbaa-1676f3c4557f)

![image](https://github.com/user-attachments/assets/e12958b0-e562-4d44-9591-43fc4b7a1e63)

**Comment by wpalaceJHA** [comment]
@dusty-phillips, thank you for writing "LazyVim for Ambitious Developers". This book was incredibly helpful in getting started with LazyVim. I just sent you a Github sponsorship to show my appreciation. If you can spare a few minutes to point me in the right direction, here, I would be tremendously helpful.

**Reply by dusty-phillips** [reply]
Try `:TSInstall powershell`

<img width="1180" alt="Screenshot 2024-10-18 at 3 37 06 PM" src="https://github.com/user-attachments/assets/e816a70c-b1c4-4a1e-ad80-db941d2e94f4">

**Reply by wpalaceJHA** [reply]
That worked. Thanks a ton!
